api/envoy/api/v2/core/health_check.proto

```diff
@@ -198,12 +199,12 @@ message HealthCheck {
   // The number of unhealthy health checks required before a host is marked
   // unhealthy. Note that for *http* health checking if a host responds with 503
   // this threshold is ignored and the host is considered unhealthy immediately.
-  google.protobuf.UInt32Value unhealthy_threshold = 4;
+  google.protobuf.UInt32Value unhealthy_threshold = 4 [(validate.rules).message = {required: true}];
```

Comment:
Strengthening PGV annotations is a quasi-breaking change (not at the proto wire level, but semantically). Was this "required" attribute always implied, and we're just documenting it better, or is this a new thing?

Reply:
My goal was initially just to document that these are required, since we discovered it the hard way.

I considered adding an annotation that would trigger the required notation in the API docs, but realized that `PROTOBUF_GET_WRAPPED_REQUIRED` (in `HealthCheckerImplBase`) makes these fields de facto required. Based on that I decided to just explicitly add the validation. I think the only difference in behavior is the exact error message generated.

The switch to `PROTOBUF_GET_WRAPPED_REQUIRED` occurred in 7786e393670e08622d6990c054703668384c28ed (circa August 2017). So it seems to me this has been required since v2.